Keen.NetStandard/KeenClient.cs

```diff
@@ -972,5 +973,177 @@ public JObject Query(string queryName, Dictionary<string, string> parms)
                 throw ex.TryUnwrap();
             }
         }
+
+        /// <summary>
+        /// Get query results from a Cached Dataset.
+        /// </summary>
+        /// <param name="datasetName">Name of cached dataset to query.</param>
+        /// <param name="indexBy">The string property value you want to retrieve results by.</param>
+        /// <param name="timeframe">Limits retrieval of results to a specific portion of the Cached Dataset</param>
+        /// <returns>An instance of Newtonsoft.Json.Linq.JObject containing query results and metadata defining the cached dataset.</returns>
+        public async Task<JObject> QueryDatasetAsync(string datasetName, string indexBy, string timeframe)
+        {
+            return await Datasets.GetResultsAsync(datasetName, indexBy, timeframe);
+        }
+
+        /// <summary>
+        /// Get query results from a Cached Dataset.
+        /// </summary>
+        /// <param name="datasetName">Name of cached dataset to query.</param>
+        /// <param name="indexBy">The string property value you want to retrieve results by.</param>
+        /// <param name="timeframe">Limits retrieval of results to a specific portion of the Cached Dataset</param>
+        /// <returns>An instance of Newtonsoft.Json.Linq.JObject containing query results and metadata defining the cached dataset.</returns>
+        public JObject QueryDataset(string datasetName, string indexBy, string timeframe)
```

Comment:
Is there really a need to provide this method? If clients really want to block they can call `.Result` on the async method. Same comment for all the other non-async methods here that call `.Result` on an async method.

Reply:
Using `.Result` in Asp.Net requests that require the http context can deadlock the thread, this is why most C# APIs offer both a sync and async alternative. While the implementation for both the Sync and Async methods here are identical, they are safe because of the use of `.ConfigureAwait(false)`. The existence of the sync method implicitly informs the person consuming `KeenClient` that the method is safe. Using `.Result` on a foreign code base from within an ASP.Net application is considered a risk.

That being said, QueryDatasetAsync is missing `ConfigureAwait(false)`, I'll add the appropriate comments there and anywhere else I can find.

Reply:
Yeah anything that handles the sync context the way ASP does is subject to this if you haven't configured the awaiter, so this is at least self-documenting in that sense. I know S. Toub would have some words  on the subject, but I think this should stay. The real issue I have here is that I think debugging is a little tough with the exceptions we get in these wrappers, which I've been meaning to look into.

Reply:
Makes sense!